### PR TITLE
Exercise new @produces, @consumes, and @supports_config decorators

### PR DESCRIPTION
**Pull request by knoepfel**
This commit adopts the decorator interface provided by the `decisionengine`'s new produces/consumes/supports_config facility.  The most significant changes include:

- Substantial boilerplate removal for supporting module program-options
- Removal of hard-coded produces/consumes information in the Python modules

**No** changes should be required in the configuration files.  This PR also removes the following files, which were no longer necessary:

- AWS/AWSSpotPrice.py
- AWS/CloudBudget.py
- AWS/CloudRequests.py
- AWS/HPCAllocation.py
- AWS/HTCondorJobQ.py
- AWS/ProvisionerResourceList.py
- glideinwms/sources/fe_config.py

### CODE REVIEW

**Comment by mambelli** [comment]
Some comment while I'm going through the files. 
1. A note to remember to add the files that Steve asked to remove in the comment.
2. When do you use dataframe and when data_block? I see that sometimes data_block (or datablock) was replaced, e.g. in AWS/publishers/AWS_figure_of_merit.py, sometimes it is left, e.g. AWS/transforms/FigureOfMerit.py. In the specs we talk about DataBlock for the data shared within a channel. Is dataframe the same?

**Review by mambelli — COMMENTED** [review]
When do you use dataframe and when data_block? I see that sometime data_block was replaced.

File: glideinwms/resource_dist_plugins.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 logger = logging.getLogger()
 
 @six.add_metaclass(abc.ABCMeta)
-class ResourceOrderPlugin(object):
+class ResourceOrderPlugin:
```
[the remark this answers is not in the document]
Comment (knoepfel) [reply glideinwms/resource_dist_plugins.py:9]: Yes, that would be a suitable replacement--I wasn't touching dependencies in this PR, but this is a simple fix.

File: htcondor/htcondor_query.py [hunk 2]
```diff
@@ -19,7 +19,7 @@ def __init__(self, err_str):
 
 
 @six.add_metaclass(abc.ABCMeta)
-class Query(object):
+class Query:
```
[the remark this answers is not in the document]
Comment (knoepfel) [reply htcondor/htcondor_query.py:22]: Will do.

File: htcondor/sources/source.py [hunk 3]
```diff
@@ -2,11 +2,20 @@
...
+
+@Source.supports_config(Parameter('collector_host', type=str),
+                        Parameter('condor_config', type=str),
+                        Parameter('constraint', default=True),
+                        Parameter('classad_attrs', type=list),
+                        Parameter('group_attr', default=['Name']),
+                        Parameter('subsystem_name', type=str),
+                        Parameter('correction_map', default={}))
 @six.add_metaclass(abc.ABCMeta)
 class ResourceManifests(Source.Source):
```
[the remark this answers is not in the document]
Comment (knoepfel) [reply htcondor/sources/source.py:20]: Understood.

File: NERSC/sources/NerscAllocationInfo.py [hunk 4]
```diff
@@ -88,80 +83,16 @@ def acquire(self):
...
-        with open(args.acquire_with_config, 'r') as f:
-            config_string = "".join(f.readlines())
-            config = literal_eval(config_string)
-        n = NerscAllocationInfo(config['sources']['NerscAllocationInfo']
-                                ['parameters'])
-        print(n.acquire())
-    else:
-        pass
 
+#    elif args.acquire_with_config:
```
[the remark this answers is not in the document]
Comment (knoepfel) [reply NERSC/sources/NerscAllocationInfo.py:89]: That comment should have been removed.  Will remove it.

**Comment by knoepfel** [comment]
> When do you use dataframe and when data_block? I see that sometime data_block was replaced.

There were several instances when the name of the function parameter was called `data_block`, but what was actually passed in was a `dataframe`: the datablock contains all data products; only *one* of those data products was being passed to the `graphite_context` method, and that product is a dataframe.

To show the difference, any publisher inheriting from (e.g.) `AWS_generic_publisher`, such as `AWS_price_performance` will specify `@consumes_dataframe` instead of `@consumes`.

**Review by mambelli — APPROVED** [review]

**Pull request merged**